### PR TITLE
 #161509243 Restore deleted travel checklist items

### PR DESCRIPTION
**Pull request by ssewilliam**
#### What does this PR do?

- Allows an Admin to restore previously deleted checklist items

#### Description of Task to be completed?

- Admin should be able to restore a disable(deleted) travel checklist item
So that it can be made visible to potential requesters.

#### How should this be manually tested?
- Clone the repo - git clone https://github.com/andela/travel_tool_front.git
- Checkout to this branch - git checkout ft-restore-deleted-items-161509243
- Install dependencies and start the server ```yarn start```
- Try to delete a checklist item and attempt to restore the same item.
- If the item is restored without page reload then the feature works well.

#### Any background context you want to provide?
- The errors flagged by codeClimate are prop-type validation errors which are for prop types that have similar structures but are not the same in any way

#### What are the relevant pivotal tracker stories?

[#161509243](https://www.pivotaltracker.com/story/show/161509243)

#### Screenshots (if appropriate)
ezgif com-optimize

### CODE REVIEW

File: src/views/Checklist/index.js [hunk 1]
```diff
@@ -28,28 +32,20 @@ export class Checklist extends Component {
     this.setState({ checklistItemId });
     openModal(true, 'delete checklist item');
   }
-  openAddModal = () => {
-    let { openModal } = this.props;
-    openModal(true, 'add cheklistItem');
+  setItemToRestore = (checklistItemId) => () => {
+    const { openModal, deletedChecklistItems } = this.props;
+    const restoreItemData = deletedChecklistItems.find(x => x.id === checklistItemId);
```
Comment (Nedson202) [inline src/views/Checklist/index.js]: where you have `x => x.id`, please update with a descriptive name

Comment (ssewilliam) [reply src/views/Checklist/index.js]: Thanks @Nedson202

File: src/components/Forms/DeleteRequestForm/FormFieldsets/SubmitArea.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import {PropTypes} from 'prop-types';
+
+const SubmitArea = (props) => {
```
Comment (DennisWanjiru) [inline src/components/Forms/DeleteRequestForm/FormFieldsets/SubmitArea.js]: You could de-structure the props directly in the args as follows  `const SubmitArea = ( {onCancel, deleteChecklistItem }) => ...`

File: src/components/Forms/DeleteRequestForm/DeleteRequestForm.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import error from '../../../images/error.svg';
+import Modal from '../../modal/Modal';
+import SubmitArea from './FormFieldsets/SubmitArea';
+
+
+const DeleteRequestForm = (props) => {
+  const { closeModal, deleteChecklistItem, shouldOpen, modalType, handleInputChange } = props;
```
Comment (DennisWanjiru) [inline src/components/Forms/DeleteRequestForm/DeleteRequestForm.js]: here too ... directly de-structure the props as shown below

File: src/components/modal/RestoreChecklistModal/RestoreChecklistModal.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        closeModal={closeModal}
+        customModalStyles="delete-checklist-item restore-model-content" visibility={
+          shouldOpen && modalType.match('restore checklist item') ? 'visible' : 'invisible'
+        }
+        title="Restore Checklist Item"
+      > 
+        <span className="delete-checklist-item__disclaimer restore-checklist-items_span">
+          Are you sure you want to restore
+          <strong>
+            &nbsp;{ itemName }&nbsp;
```
Comment (DennisWanjiru) [inline src/components/modal/RestoreChecklistModal/RestoreChecklistModal.js]: you could style your elements instead of using non-breaking space entity.

**Pull request merged**